File: src/Client/Logistics.OfficeApp/src/app/core/api/models/load/create-load.command.ts
import {AddressDto} from "../address.dto";
import {GeoPointDto} from "../geo-point.dto";
import {LoadType} from "./enums";

export interface CreateLoadCommand {
  name: string;
  loadType: LoadType;
  originAddress: AddressDto;
  originLocation: GeoPointDto;
  destinationAddress: AddressDto;
  destinationLocation: GeoPointDto;
  deliveryCost: number;
  distance: number;
  assignedDispatcherId: string;
  assignedTruckId: string;
  customerId?: string;
}

File: src/Core/Logistics.Application/Commands/Load/CreateLoad/CreateLoadCommand.cs
using Logistics.Application.Abstractions;
using Logistics.Domain.Primitives.Enums;
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Application.Commands;

public class CreateLoadCommand : IAppRequest
{
    public string Name { get; set; } = null!;
    public LoadType Type { get; set; }
    public Address OriginAddress { get; set; } = null!;
    public GeoPoint OriginLocation { get; set; } = null!;
    public Address DestinationAddress { get; set; } = null!;
    public GeoPoint DestinationLocation { get; set; } = null!;
    public decimal DeliveryCost { get; set; }
    public double Distance { get; set; }
    public Guid AssignedDispatcherId { get; set; }
    public Guid? AssignedTruckId { get; set; }
    public Guid CustomerId { get; set; }

    public LoadSource Source { get; set; } = LoadSource.Manual;
    public DateTime? RequestedPickupDate { get; set; }
    public DateTime? RequestedDeliveryDate { get; set; }
    public string? Notes { get; set; }
    public Guid? ContainerId { get; set; }
    public Guid? OriginTerminalId { get; set; }
    public Guid? DestinationTerminalId { get; set; }
}
